tts-evaluation-MCD-DTW.ipynb

```diff
@@ -0,0 +1,501 @@
+   "metadata": {},
+   "source": [
+    "# Calculating MCD DTW(Mel cepstral distortion - Dynamic time warping)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8a749b10",
+   "metadata": {},
+   "source": [
+    "In this tutorial we are going to learn how to calculate MCD DTW between a synthesized audio and reference audio.\n",
+    "\n",
+    "Mel cepstral distortion(MCD) is an objective measure of speech quality. MCD is calculated between a tts generated audio and a ground truth audio. Two audios are similar, if the MCD value between them is low. MCD of an audio with itself is 0.\n",
+    "\n",
+    "MCD DTW is a modification of MCD that works with non aligned audio clips by using dynamic time warping cost matrix. The scale depends on factors such as mel extractor and reduction algorithm(mean of DTW cost or min DTW path cost). It is useful for comparing the model convergence trained on same training data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "09cff1ab",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Install necessary packages\n",
+    "!pip install librosa numpy matplotlib"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8b9b6e1c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## Import libraries\n",
+    "import librosa\n",
+    "import librosa.display\n",
+    "import numpy as np\n",
+    "import math\n",
+    "import matplotlib.pyplot as plt\n",
+    "import os"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "398b01e0",
+   "metadata": {},
+   "source": [
+    "### Define parameters for spectrogram generation.\n",
+    "In this section we define parameters required to generate our spectrograms. More information about these parameteres can found on librosa documentation for [stft](https://librosa.org/doc/main/generated/librosa.stft.html), [mels](https://librosa.org/doc/main/generated/librosa.filters.mel.html) and [mfcc](https://librosa.org/doc/main/generated/librosa.feature.mfcc.html) generation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "191012de",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## Mel params\n",
+    "n_fft=2048\n",
+    "hop_length=512\n",
+    "win_length=None\n",
+    "window='hann'\n",
+    "n_mels = 34\n",
+    "\n",
+    "## Mfcc params\n",
+    "n_mfcc=n_mels"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "54830c8c",
+   "metadata": {},
+   "source": [
+    "A little bit about these parameters:  \n",
+    "    - `n_fft`: Number of fft_components or length of windowed signal after padding in stft.  \n",
+    "    - `hop_length`: Number of audio samples between adjacent STFT columns.  \n",
+    "    - `window`: Window to use in stft.  \n",
+    "    - `win_length`: Length of the window to be used.  \n",
+    "    - `n_mels`: Number of number of Mel bands to generate.\n",
+    "    - `n_mfcc`: Number of MFCCs to generate."
```

Comment:
This doesn't render properly on https://github.dev/styagi130/tutorials/blob/tutorial__mcd/tts-evaluation-MCD-DTW.ipynb. Can you fix?